In [1]:
import pandas as pd
df=pd.read_csv('https://raw.githubusercontent.com/2deezer2/-Auto-pricing-mashinki-/main/Data.csv')
df

Unnamed: 0  Custom  Number_of_owners  Year      Price  Region  \
0               0       1                 4  2000   310000.0  Москва   
1               1       1                 1  2022  3500000.0  Москва   
2               2       1                 1  2019  8000000.0  Москва   
3               3       1                 2  2011   310000.0  Москва   
4               4       1                 3  2018  2020000.0  Москва   
...           ...     ...               ...   ...        ...     ...   
66928       66928       1                 3  2013  2490000.0   Химки   
66929       66929       1                 1  2019  9700000.0  Москва   
66930       66930       1                 1  2018  5250000.0  Москва   
66931       66931       1                 3  2013  2880000.0  Москва   
66932       66932       1                 1  2021  6620000.0  Москва   

        Mileage          Brand  Сегмент  Power               Type Engine_type  \
0      132000.0     LADA (ВАЗ)  ECONOMY   79.0  Внедорожник 3 дв.    GASOLINE   
1        4000.0          Skoda   MEDIUM  190.0            Лифтбек    GASOLINE   
2       14000.0  Mercedes-Benz  PREMIUM  469.0         Седан Long    GASOLINE   
3       25000.0     LADA (ВАЗ)  ECONOMY   80.0              Седан    GASOLINE   
4       61000.0           Audi  PREMIUM  150.0              Седан    GASOLINE   
...         ...            ...      ...    ...                ...         ...   
66928  120000.0     Land Rover  PREMIUM  245.0  Внедорожник 5 дв.      DIESEL   
66929   61000.0     Land Rover  PREMIUM  380.0  Внедорожник 5 дв.    GASOLINE   
66930  159000.0            BMW  PREMIUM  265.0  Внедорожник 5 дв.      DIESEL   
66931   96000.0            BMW  PREMIUM  184.0               Купе      DIESEL   
66932    7000.0           Jeep   MEDIUM  470.0  Внедорожник 5 дв.    GASOLINE   

      Положение руля           Привод Transmission  
0               LEFT  ALL_WHEEL_DRIVE   MECHANICAL  
1               LEFT  FORWARD_CONTROL        ROBOT  
2               LEFT  ALL_WHEEL_DRIVE    AUTOMATIC  
3               LEFT       REAR_DRIVE   MECHANICAL  
4               LEFT  FORWARD_CONTROL        ROBOT  
...              ...              ...          ...  
66928           LEFT  ALL_WHEEL_DRIVE    AUTOMATIC  
66929           LEFT  ALL_WHEEL_DRIVE    AUTOMATIC  
66930           LEFT  ALL_WHEEL_DRIVE    AUTOMATIC  
66931           LEFT  ALL_WHEEL_DRIVE   MECHANICAL  
66932           LEFT  ALL_WHEEL_DRIVE    AUTOMATIC  

[66933 rows x 15 columns]

## **Чем новее модель автомобиля, тем выше цена на него.**

Для проверки данной гипотезы можно использовать корреляционный анализ.

In [9]:
import numpy as np
from scipy.stats import pearsonr

df['Age'] = 2023 - df['Year']

# вычисление коэффициента корреляции Пирсона
corr, _ = pearsonr(df['Age'], df['Price'])

# вывод результата
if corr > 0:
    print('Есть положительная корреляция между возрастом модели и ценой на автомобиль')
else:
    print('Есть отрицательная корреляции между возрастом модели и ценой на автомобиль')
print(f'Коэффициент корреляции: {corr:.2f}')

Есть отрицательная корреляции между возрастом модели и ценой на автомобиль
Коэффициент корреляции: -0.56


Для определения статистической значимости и проверки нулевой гипотезы (нет корреляции между возрастом модели и ценой на автомобиль) можно использовать t-тест Стьюдента.

In [8]:
from scipy.stats import t
from numpy import sqrt

# вычисление длины выборки
n = len(df)

# вычисление значения t-статистики
t_stat = corr * sqrt(n - 2) / sqrt(1 - corr**2)

# определение уровня значимости для t-теста
alpha = 0.05

# определение критической области
cv = t.ppf(1 - alpha / 2, n - 2)

# вывод результата
if abs(t_stat) > cv:
    print('Корреляция между возрастом модели и ценой на автомобиль является статистически значимой')
else:
    print('Корреляция между возрастом модели и ценой на автомобиль не является статистически значимой')
print(f't-статистика: ', t_stat)

Корреляция между возрастом модели и ценой на автомобиль является статистически значимой
t-статистика:  -176.61363417271056


Корреляция между возрастом модели и ценой на автомобиль является статистически значимой, поэтому мы можем сделать вывод о том, что существует значимая связь между возрастом автомобиля и его ценой. В данном случае, коэффициент корреляции отрицательный, соответственно можно сделать вывод о том, что на рынке автомобилей более старших лет цены ниже, чем на более новые модели.

Вывод: **гипотеза подтвердилась**.

##**Цена на автомобиль тем меньше, чем больше его пробег**

In [10]:
corr, pval = pearsonr(df['Mileage'], df['Price'])

# Вывод результатов корреляционного анализа
print(f'Коэффициент корреляции: {corr:.3f}')
print(f'P-value: {pval:.3f}')

if pval < 0.05:
    print('Существует статистически значимая зависимость между ценой и пробегом')
else:
    print('Нет статистически значимой зависимости между ценой и пробегом')

Коэффициент корреляции: -0.482
P-value: 0.000
Существует статистически значимая зависимость между ценой и пробегом


Так как значение P-value меньше 0.05, то существует статистически значимая зависимость между ценой и пробегом, а коэффициент корреляции Пирсона -0.482 говорит о том, что присутствует отрицательная корреляционная связь между ценой и пробегом, то есть чем больше пробег, тем ниже цена на автомобиль. Вывод: **гипотеза подтвердилась.**

## **Автомобили, которые имеют наменьшее количество владельцев, стоят дороже.**

In [17]:
corr, pval = pearsonr(df['Number_of_owners'], df['Price'])

# Вывод результатов корреляционного анализа
print(f'Коэффициент корреляции: {corr:.3f}')
print(f'P-value: {pval:.3f}')

if pval < 0.05:
    print('Существует статистически значимая зависимость между ценой и количеством владельцев')
else:
    print('Нет статистически значимой зависимости между ценой и количеством владельцев')

Коэффициент корреляции: -0.584
P-value: 0.000
Существует статистически значимая зависимость между ценой и пробегом


Исходя из полученных результатов, можно сделать вывод, что между ценой и пробегом существует отрицательная связь. Коэффициент корреляции равен -0,584, что говорит о том, что с увеличением пробега цена на автомобиль снижается.

P-value, равное 0, означает, что вероятность получить такой или еще более выраженный коэффициент корреляции при отсутствии связи между переменными крайне мала, что говорит о статистической значимости полученной зависимости.

Таким образом, можно сделать вывод, что между ценой и пробегом существует статистически значимая зависимость, и пробег может быть одним из факторов, влияющих на цену автомобиля. Вывод: **гипотеза подтвердилась**